engine/src/main/java/org/terasology/engine/input/lwjgl/LwjglMouseDevice.java

```diff
@@ -123,6 +117,21 @@ public void propertyChange(PropertyChangeEvent evt) {
         }
     }
 
+    private void focusCallback(long window, boolean isFocused) {
+        // When the window is unfocused and focused again, the mouse coordinates
+        // reported by GLFW change even if the mouse hasn't moved.
+        // So when the window regains focus, we set the new position as the reference without generating deltas.
+        if (isFocused) {
+            DoubleBuffer mouseX = BufferUtils.createDoubleBuffer(1);
+            DoubleBuffer mouseY = BufferUtils.createDoubleBuffer(1);
+
+            GLFW.glfwGetCursorPos(window, mouseX, mouseY);
+
+            xpos = mouseX.get(0);
+            ypos = mouseY.get(0);
```

Comment:
can this be replaced with a Vector2d;

Reply:
Yes, it can. That does make it shorter, as long as we're fine with `newPos.sub(pos, posDelta)` being somewhat harder to read than `xposDelta = x - this.xpos`.

Reply:
It would also require corresponding changes to the `update` method which uses those fields.

I know I've encouraged a bit of scope creep here by bringing up the code duplication in `update` and the use of `rewind`, but we should probably not stack _all_ of the possible clean-ups we can think of in to this bugfix PR.